### PR TITLE
Remove the need for an AST transform

### LINKED ISSUE #81: Can we drop the ast transform?
I don't understand why this addon uses an AST transform to rewrite `element` to `-element`.

If it's an attempt at future-compatibility with a glimmer-native implementation, that seems defeated by the fact that this addon already ships a helper named `element` anyway. So why not use that helper and eliminate the need for a custom transform?

The motivation for eliminating the transform is that we do not intend to let v2 addons automatically install AST transforms, so if this is kept it's going to become an extra manual step that users of the addon will need to deal with.

### PR DESCRIPTION
**Pull request by SergeAstapov**
Closes #81 cc @ef4

The only failing test case is `it can be invoked inline` where helper is used like so:

```js
this.set('tagName', 'p');
await render(hbs`{{element this.tagName}}`);
```

it renders `[Object object]` instead of rendering the component.

Would appreciate if anyone would be able to point out how `{{component}}` template helper could be invoked programmatically or how to wrap result of `ensureSafeComponent()` function into invocable reference.

### CODE REVIEW

File: tests/integration/helpers/element-test.js
Comment (SergeAstapov) [inline tests/integration/helpers/element-test.js:242]: As discussed in team meeting today with @ef4 and @rwjblue, this failing test reveals an issue in Ember or Glimmer VM itself and will be tracked separately.

File: tests/integration/helpers/element-test.js
Comment (SergeAstapov) [inline tests/integration/helpers/element-test.js:131]: This is a breaking change in this PR:
AST was transforming this line

```js
{{#let (component (element "h1")) as |Tag|}}
```

into

```js
{{#let (component (component (ensure-safe-component (element "h1")))) as |Tag|}}
```

which *implicitly* Embroider compatible but now requires users explicitly use `ensure-safe-component`.

Comment (rwjblue) [reply tests/integration/helpers/element-test.js:131]: We _could_ use the JS api for ensureSafeComponent if we wanted to keep this behavior, but I’m not really super clear on what value that adds for us. 

Why would folks need/want to do this wrapping manually?

Comment (rwjblue) [reply tests/integration/helpers/element-test.js:131]: I’m guessing that this is specifically to support passing the test under Ember 3.24 only?

https://github.com/tildeio/ember-element-helper/blob/84167f34a33743d7a5abf966d15237604a3f6c11/.github/workflows/ci.yml#L61

On Ember >= 3.25 you can pass a component class directly to {{component without using ensureSafeComponent…

Comment (SergeAstapov) [reply tests/integration/helpers/element-test.js:131]: @rwjblue Actually I already use JS api for `ensureSafeComponent` in the `element` helper:

https://github.com/SergeAstapov/ember-element-helper/blob/b8ed665cfc49b6b77dd5b97932ad48e40ae7b1fd/addon/helpers/element.js#L32-L37

The failing scenario actually was `embroider-optimized`, not `ember-lts-3.24`.
The error was

> $TMPDIR/embroider/efb531/tests/dummy/tests/integration/helpers/element-test.js: Unsafe dynamic component: cannot statically analyze this expression in tests/integration/helpers/element-test.js

@rwjblue you can see test failure with detailed logs here https://github.com/SergeAstapov/ember-element-helper/runs/5414404831?check_suite_focus=true

I assume Embroider does not like usage like `(component (some-helper))` as it assumes dynamic invocation and can't statically resolve the result of helper invocation and requires to wrap result of `element` helper into `ensure-safe-component` helper.

Not sure what I'm missing here.

Comment (rwjblue) [reply tests/integration/helpers/element-test.js:131]: I see, thank you! I do think that is something we could possibly look into on the Embroider side (though @ef4 would have a better mental model than me), but that doesn't need to block this PR.

One thing that we should also do is explicitly drop support for Ember < 3.24, because I think you'd **have** to use `ensureSafeComponent` in a template like this under 3.24:

```hbs
{{#let (element 'h1') as |Thing|}}
  <Thing>Hello!</Thing>
{{/let}}
```

Because you aren't allowed (until 3.25) to directly invoke a compoennt class. You'd have to do something like this under 3.24:

```hbs
{{#let (ensure-safe-component (element 'h1')) as |Thing|}}
  <Thing>Hello!</Thing>
{{/let}}
```

Comment (SergeAstapov) [reply tests/integration/helpers/element-test.js:131]: @rwjblue sorry, did you mean "One thing that we should also do is explicitly drop support for Ember < 3.25" (with 3.25 instead of 3.24)?

I'm totally fine to do that, just for the context planned next step once this is merged - convert addon to v2 format.
So we can bundle these breaking changes into single release

Comment (rwjblue) [reply tests/integration/helpers/element-test.js:131]: After discussing with @ef4, since the implementation of `addon/helpers/element.js` wrap the result in `ensureSafeComponent` already things should work just fine in "normal" cases. The only scenario that will break (as you pointed out) is when you explicitly pass `{{component (element` **and** are using `embroider-optimized`. 

I'd even suggest that we don't consider it a breaking change, as I don't _really_ think folks would do `{{component (element 'div')}}`.

Comment (rwjblue) [reply tests/integration/helpers/element-test.js:131]: After discussing with @ef4, since the implementation of `addon/helpers/element.js` wrap the result in `ensureSafeComponent` already things should work just fine in "normal" cases. The only scenario that will break (as you pointed out) is when you explicitly pass `{{component (element` **and** are using `embroider-optimized`. 

I'd even suggest that we don't consider it a breaking change, as I don't _really_ think folks would do `{{component (element 'div')}}`.

**Review by rwjblue — APPROVED** [review]

**Comment by wagenet** [comment]
This change seems to break ember-animated: https://github.com/ember-animation/ember-animated/issues/450

**Comment by grodriguez85** [comment]
This change also seems to break ember-basic-dropdown

**Comment by SergeAstapov** [comment]
> This change also seems to break ember-basic-dropdown

@grodriguez85 could you please provide more info?

**Comment by grodriguez85** [comment]
I did some debugging and found the bug was on my side. On an addon I'm developing we use ember-power-select, which ends using ember-element-helper v0.6.1. But on my addon I was directly using ember-element-helper on the v0.5.5. So the AST wasn't transforming anything but the code from v0.5.5 was stuck on this [assertion](https://github.com/tildeio/ember-element-helper/blob/v0.5.5/addon/helpers/element.js).

**Pull request merged**
